src/AbstractCollection.php

```diff
@@ -57,9 +62,162 @@ public function remove($element): bool
         if (($position = array_search($element, $this->data, true)) !== false) {
             unset($this->data[$position]);
 
+
             return true;
         }
 
         return false;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function column(string $propertyOrMethod): array
+    {
+        $temp = [];
+
+        foreach ($this->data as $item) {
+            $temp[] = $this->extractValue($item, $propertyOrMethod);
+        }
+
+        return $temp;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function first()
+    {
+        if (empty($this->data)) {
+            throw new OutOfBoundsException('Can\'t determine first item. Collection is empty');
+        }
+
+        reset($this->data);
```

Comment:
Should this be using `\reset()` to ensure top-level namespace?

Reply:
Will set it up for the functions I see. Good point. Thans

Reply:
Actually, did some digging and this does not count for every functions there is.
Only a specific set of functions will take usage of a better opcode when calling them in root namespace.

See here for the list.
https://github.com/php/php-src/blob/master/Zend/zend_compile.c#L3803

So adding `\reset()` does not bring any benefit.

Reply:
Wouldn't it be better to always use root namespace for consistency? Even if there is no benefit now, maybe there will be at a later date.

Reply:
I think this should be discussed. I would like to add those with opcode benefit in the files I changed. For the other files I would file a new PR where we can discuss this with everyone else on a different matter.

Reply:
IMO trying to optimize at this level is pretty meaningless. There *is* an argument to made for the security of root-ing all functions to prevent hijacking.

Reply:
Filed an issue for this. #46